mesonbuild/backend/ninjabackend.py

```diff
@@ -2013,7 +2013,8 @@ def get_custom_target_dir_include_args(self, target, compiler):
             # own target build dir.
             if not isinstance(i, (build.CustomTarget, build.CustomTargetIndex)):
                 continue
-            idir = self.get_target_dir(i)
+            # NOTE: if we want to use os.path.realpath() for include path only! CK
+            idir = os.path.normpath(self.get_target_dir(i))
```

Comment:
The comment does not make sense. Also, please don't add initials or any such tags to the code. They are just noise, that's what `git blame` is for. The same goes for the other two comments below

It would probably also be better to make `get_target_dir` return normalised paths. This way it fixes the issue for all call sites.

Reply:
see https://github.com/ninja-build/ninja/issues/1251

Reply:
done